DuckDuckGo/EmailWaitlistViewController.swift

```diff
@@ -0,0 +1,388 @@
+        didSet {
+            applyWaitlistButtonStyle(to: existingInviteCodeButton)
+        }
+    }
+
+    @IBOutlet weak var existingDuckAddressButton: UIButton! {
+        didSet {
+            applyWaitlistButtonStyle(to: existingDuckAddressButton)
+        }
+    }
+
+    lazy var emailManager: EmailManager = {
+        let emailManager = EmailManager()
+        emailManager.requestDelegate = self
+        return emailManager
+    }()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        applyTheme(ThemeManager.shared.currentTheme)
+        renderCurrentWaitlistState()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        replaceViewControllerWithSignedInStateIfNeeded()
+    }
+
+    @IBAction func waitlistActionButtonTapped(_ sender: UIButton) {
+        switch emailManager.waitlistState {
+        case .notJoinedQueue: joinWaitlist()
+        case .inBeta: getStarted()
+        case .joinedQueue:
+            assertionFailure("\(#file): No action for .joinedQueue")
+            return
+        }
+    }
+
+    @IBAction func existingInviteCodeButtonTapped(_ sender: UIButton) {
+        showEmailWaitlistWebViewController(url: AppUrls().signUpQuickLink)
+    }
+
+    @IBAction func existingDuckAddressButtonTapped(_ sender: UIButton) {
+        showEmailWaitlistWebViewController(url: AppUrls().emailLoginQuickLink)
+    }
+
+    // MARK: - Private
+
+    private func textViewLinkAttributes() -> [NSAttributedString.Key: Any] {
+        let paragraphStyle = NSMutableParagraphStyle()
+        paragraphStyle.lineHeightMultiple = 1.16
+
+        let linkAttributes: [NSAttributedString.Key: Any] = [
+            NSAttributedString.Key.paragraphStyle: paragraphStyle,
+            NSAttributedString.Key.font: 16,
+            NSAttributedString.Key.foregroundColor: UIColor.emailWaitlistLinkColor
+        ]
+
+        return linkAttributes
+    }
+
+    private func applyWaitlistButtonStyle(to button: UIButton, hasSolidBackground: Bool = false) {
+        button.layer.cornerRadius = button.bounds.height / 2
+        button.layer.borderWidth = 2.0
+        button.layer.borderColor = UIColor.emailWaitlistLinkColor.cgColor
+
+        if hasSolidBackground {
+            button.layer.backgroundColor = UIColor.emailWaitlistLinkColor.cgColor
+        }
+    }
+
+    private func replaceViewControllerWithSignedInStateIfNeeded() {
+        if emailManager.isSignedIn, let existingViewControllers = navigationController?.viewControllers {
+            let storyboard = UIStoryboard(name: "Settings", bundle: Bundle.main)
+            let viewController = storyboard.instantiateViewController(identifier: "EmailProtectionViewController")
+
+            let newViewControllers = existingViewControllers.dropLast() + [viewController]
+            navigationController?.setViewControllers(Array(newViewControllers), animated: false)
+        }
```

Comment:
This would be better suited to having multiple child view controllers, but the waitlist classes are going to be completely removed later on, so I'm hesitant to embed them into the app too deeply.

That said, this `setViewControllers` dance could be changed to instead pop to the root view controller and push the protection VC back on instead. It's just a quick way to check that the right VC is displayed if the user is signed in.